infrastructure/repository/contact_repository_test.go

```diff
@@ -0,0 +1,291 @@
+	repo, contact := setupContactTest(t)
+	defer cleanupContactTest(t, repo, contact)
+
+	createdContact, err := repo.GetContactByUsername(*contact)
+	assert.NoError(t, err)
+	assert.Equal(t, contact.UserID, createdContact.UserID)
+	assert.Equal(t, contact.PhoneBookID, createdContact.PhoneBookID)
+	assert.Equal(t, contact.FirstName, createdContact.FirstName)
+	assert.Equal(t, contact.LastName, createdContact.LastName)
+	assert.Equal(t, contact.Username, createdContact.Username)
+	assert.Equal(t, contact.PhoneNumber, createdContact.PhoneNumber)
+}
+
+// TestUpdateContactByUsername tests updating a contact by username.
+func TestUpdateContactByUsername(t *testing.T) {
+	repo, contact := setupContactTest(t)
+	defer cleanupContactTest(t, repo, contact)
+
+	// Invalid contact with a new username
+	invalidContact := entities.Contact{
+		UserID:      contact.UserID,
+		Username:    uuid.NewString(),
+		FirstName:   "Amir Hossein",
+		LastName:    "-T",
+		PhoneNumber: "09100000000",
+	}
+
+	_, err := repo.UpdateContactByUsername(invalidContact)
+	assert.Error(t, err)
+
+	// Edited contact with updated information
+	editedContact := entities.Contact{
+		UserID:      contact.UserID,
+		Username:    contact.Username,
+		FirstName:   "Amir Hossein",
+		LastName:    "-T",
+		PhoneNumber: "09100000000",
+	}
+
+	updatedContact, err := repo.UpdateContactByUsername(editedContact)
+	assert.NoError(t, err)
+	assert.Equal(t, editedContact.UserID, updatedContact.UserID)
+	assert.Equal(t, editedContact.Username, updatedContact.Username)
+	assert.Equal(t, editedContact.FirstName, updatedContact.FirstName)
+	assert.Equal(t, editedContact.LastName, updatedContact.LastName)
+	assert.Equal(t, editedContact.PhoneNumber, updatedContact.PhoneNumber)
+}
+
+// TestDeleteContactUsername tests deleting a contact by username.
+func TestDeleteContactUsername(t *testing.T) {
+	repo, contact := setupContactTest(t)
+	defer cleanupContactTest(t, repo, contact)
+
+	err := repo.DeleteContactByUsername(*contact)
+	assert.NoError(t, err)
+
+	var deletedContact entities.PhoneBook
+	err = repo.DB.Where("user_id = ? AND username = ?", contact.UserID, contact.Username).First(&deletedContact).Error
+	assert.Error(t, err)
+}
+
+// TestGetContactList tests getting a list of contacts.
+func TestGetContactList(t *testing.T) {
+	repo, contact := setupContactTest(t)
+	defer cleanupContactTest(t, repo, contact)
+
+	// Contact with invalid UserID
+	contactWithInvalidUserID := entities.Contact{
+		UserID:      uuid.New(),
+		PhoneBookID: contact.PhoneBookID,
+	}
+	contacts, err := repo.GetContactList(contactWithInvalidUserID)
+	assert.NoError(t, err)
+	assert.NotNil(t, contacts)
+	assert.Empty(t, err)
+
+	// Contact with invalid PhoneBookID
+	contactWithInvalidPhoneBookID := entities.Contact{
+		UserID:      contact.UserID,
+		PhoneBookID: uuid.New(),
+	}
+	contacts, err = repo.GetContactList(contactWithInvalidPhoneBookID)
+	assert.NoError(t, err)
+	assert.NotNil(t, contacts)
+	assert.Empty(t, err)
```

Comment:
it appears that these are testing functionality with Invalid entries, shouldn't they assert Error? if these are invalid, they SHOULD ERROR and NOT be empty/noerror